.github/workflows/daily-testing.yml

```diff
@@ -0,0 +1,64 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: daily-testing
+
+on:
+  push:
+    branches:
+      - gha-testing
+  schedule:
+    # daily testing to occur at 7:30 CST every day
+    - cron:  '30 13 * * *'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
Testing all combinations of these parameters is overkill (compute time is not free). The strategy we used on Travis is to have one test with the latest of all libraries, then tests for specific older versions that require special behavior (e.g., TF v1) that we've coded for.

I'd suggest the following tests (all using the latest Python):
- [ ] Latest TF v2, no Keras
- [ ] Tensorflow v1, Keras 2.13
- [ ] Tensorflow v1, Keras 2.2.3
- [ ] Scikit-learn 0.19.1

For reference, these are the test I used in Travis. You'll need some if/then logic to handle test cases, which you can do in Bash with syntax like this: [link](https://github.com/DLHub-Argonne/dlhub_sdk/blob/master/travis/install.sh#L10)

Reply:
In the process of trying this, but I tried using the logic I saw in the document you linked, I have line `pip install keras==${{matrix.keras-version}} "tensorflow<2" "h5py<3"`

but it threw this error in the action: 
`Collecting keras==2.2.3
  Downloading Keras-2.2.3-py2.py3-none-any.whl (312 kB)
ERROR: Could not find a version that satisfies the requirement tensorflow<2 (from versions: 2.5.0rc0, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0)
ERROR: No matching distribution found for tensorflow<2
Error: Process completed with exit code 1.`

Ran into a similar issue before -- I couldn't install tensorflow versions that are less than 2.

Reply:
I think it would be better to use a specify exact versions style for defining the test loops rather than having this "do all combinations style." 

Check out QCEngine's script as an example: https://github.com/MolSSI/QCEngine/blob/master/.github/workflows/CI.yml#L16